src/Shared/Helpers/PackageDownloader.cs

```diff
@@ -212,30 +212,32 @@ public async void ClearPackageLocalCopy()
             bool doExtract)
         {
             List<string> downloadPaths = new List<string>();
-            if (metadataOnly)
+            if (packageManager != null)
             {
-                var metadata = await this.packageManager.GetMetadata(purl);
-                if (metadata != null)
+                if (metadataOnly)
                 {
-                    var outputFilename = Path.Combine(this.packageManager.TopLevelExtractionDirectory, $"metadata-{purl.ToStringFilename()}");
-
-                    // this will be effectively the same as above, if the cache doesnt exist
-                    if (!this.actualCaching)
+                    var metadata = await packageManager.GetMetadata(purl);
+                    if (metadata != null)
                     {
-                        while (File.Exists(outputFilename))
+                        var outputFilename = Path.Combine(packageManager.TopLevelExtractionDirectory, $"metadata-{purl.ToStringFilename()}");
+
+                        // this will be effectively the same as above, if the cache doesnt exist
+                        if (!this.actualCaching)
                         {
-                            outputFilename = Path.Combine(this.packageManager.TopLevelExtractionDirectory, $"metadata-{purl.ToStringFilename()}-{DateTime.Now.Ticks}");
+                            while (File.Exists(outputFilename))
+                            {
+                                outputFilename = Path.Combine(packageManager.TopLevelExtractionDirectory, $"metadata-{purl.ToStringFilename()}-{DateTime.Now.Ticks}");
+                            }
                         }
+                        File.WriteAllText(outputFilename, metadata);
+                        downloadPaths.Add(outputFilename);
                     }
-                    File.WriteAllText(outputFilename, metadata);
-                    downloadPaths.Add(outputFilename);
                 }
-            }
-            else
-            {
-                // only version download requests reach here
-                downloadPaths.AddRange(await this.packageManager.DownloadVersion(purl, doExtract, this.actualCaching));
-                this.downloadPaths = downloadPaths;
```

Comment:
I see why that was being set now.

I think we need to do this a different way.  If would call this function twice you'd lose the first download paths.  Should it be a List instead?

Reply:
I see we were already using a list.  I switched to using AddRange instead of =, so extra calls won't overwrite the download paths that already existed and added a Clear in the ClearLocal copies function.